content/en/docs/architecture/ci-operator.md

```diff
@@ -220,6 +220,27 @@ By making use of the previously compiled artifacts in the intermediate `pipeline
 cache the Go build. If multiple output `images` exist that rely on a previously built artifact, this caching effect can
 reduce build times dramatically.
 
+## Build Arguments
+The `build_args` option in `ci-operator` configuration specifies a list of [build arguments](https://docs.openshift.com/container-platform/4.7/cicd/builds/build-strategies.html#builds-strategy-docker-build-arguments_build-strategies). The values of those arguments are passed to the build at the build time to override their default values from Dockerfile. The value of a build argument can either from the field `value` in the configuration or from a referred secret. __Note__ that it is [required](/docs/how-tos/adding-a-new-secret-to-ci/#use-a-secret-in-a-job-step) to use a secret in the `test-credentials` namespace.
```

Comment:
What happens if the secret is not in that namespace? Looking at the config it should work or am I missing something?

Reply:
Correct. It will work if that namespace allows `sa/ci-operator` to read secrets. Otherwise, error about complaining about missing RBACs.

Same as the requirement in the linked page.
https://docs.ci.openshift.org/docs/how-tos/adding-a-new-secret-to-ci/#use-a-secret-in-a-job-step

Reply:
ok, so why do we allow configuring a namespace if it always must have this one value?

Reply:
My excuse is that it would be the same reason as https://docs.ci.openshift.org/docs/architecture/step-registry/#injecting-custom-credentials